infra/main.tf

```diff
@@ -582,3 +691,31 @@ resource "google_dns_record_set" "internal_gateway" {
 
   rrdatas = [google_compute_address.internal_gateway.address]
 }
+
+resource "kubernetes_cluster_role" "batch" {
```

Comment:
Can we delete batch/service-account.yaml?

Reply:
This ClusterRole and the `batch/service-account.yaml` serve different purposes. The cluster role is necessary for *default* batch because it is essentially responsible for creating and managing other namespaces. The roles in `batch/service-account.yaml` are not cluster roles, but roles within a namespace, and are applied with `kubectl -f apply -n PR_OR_DEV_NAMESPACE` and that allows the batch in a particular namespace access to the secrets and service accounts within the same namespace. This ClusterRole was added by hand years ago.